### DISCUSSION TITLE
How can I implement a knowledge base and multi-turn file dialogue with pip deployment?! How does this “history” work?

### DISCUSSION THREAD

**Discussion by SKKKKYLAR** [langfuse/langfuse#4116] [Support] [answered]
是否需要修改源代码？那我pip部署怎么操作呢？需要重新进行代码部署吗？
Do I need to modify the source code? How do I operate with pip deployment? Do I need to redeploy the code?

**Comment by marcklingen** [comment]
can you please reopen this thread in english?

**Reply by SKKKKYLAR** [reply]
ok

**Comment by marcklingen** [comment]
Thanks for translating to English. Can you provide more context regarding your question? I do not understand how it relates to Langfuse. You can use the Python SDKs via pip, and the server can run fully managed in Langfuse Cloud, or you can run the Docker container yourself.

**Reply by SKKKKYLAR** [reply]
How is multi-round conversation implemented in pip deployment？

**Reply by marcklingen** [reply, marked as answer]
I'd suggest to track this is a session: https://langfuse.com/docs/tracing-features/sessions
